src/horizontalaligner.cpp

```diff
@@ -33,6 +34,124 @@
 
 namespace vrv {
 
+//----------------------------------------------------------------------------
+// Fraction
+//----------------------------------------------------------------------------
+
+Fraction::Fraction(int num, int denom) : m_numerator(num), m_denominator(denom)
+{
+    if (denom == 0) {
+        LogDebug("Denominator cannot be zero.");
+        denom = 1;
+    }
+    Reduce();
+}
+
+Fraction::Fraction(data_DURATION duration)
+{
+    duration = vrv::DurationMin(duration, DURATION_1024);
+    duration = vrv::DurationMax(duration, DURATION_NONE);
+    int den = pow(2, (duration + 1));
+    m_numerator = 8;
+    m_denominator = den;
+    Reduce();
+}
+
+Fraction Fraction::operator+(const Fraction &other) const
+{
+    int num = m_numerator * other.m_denominator + other.m_numerator * m_denominator;
+    int denom = m_denominator * other.m_denominator;
+    return Fraction(num, denom);
+}
+
+Fraction Fraction::operator-(const Fraction &other) const
+{
+    int num = m_numerator * other.m_denominator - other.m_numerator * m_denominator;
+    int denom = m_denominator * other.m_denominator;
+    return Fraction(num, denom);
+}
+
+Fraction Fraction::operator*(const Fraction &other) const
+{
+    int num = m_numerator * other.m_numerator;
+    int denom = m_denominator * other.m_denominator;
+    return Fraction(num, denom);
+}
+
+Fraction Fraction::operator/(const Fraction &other) const
+{
+    if (other.m_numerator == 0) {
+        LogDebug("Cannot divide by zero.");
+        return *this;
+    }
+    int num = m_numerator * other.m_denominator;
+    int denom = m_denominator * other.m_numerator;
+    return Fraction(num, denom);
+}
+
+bool Fraction::operator==(const Fraction &other) const
+{
+    return (m_numerator == other.m_numerator) && (m_denominator == other.m_denominator);
+}
```

Comment:
This is weird compared to the others, because it would mean that 1/2 is not equal to 3/6. Since we now should have C++20 available, this is a neat example to try out the new spaceship operator:

```cpp
bool Fraction::operator<=>(const Fraction &other) const
 {
     return m_numerator * other.m_denominator <=> other.m_numerator * m_denominator;
 }
```

In theory, this should replace the implementation for all five operators `==`, `<`, `<=`, `>`, `>=`.

Reply:
You a right too. I guess this has not been problematic because fraction are reduced throughout the code. But it should be fixed. I was not ware of the spaceship operator! I'll change to it.